src/marine/amsr2_icec2ioda.py

```diff
@@ -0,0 +1,134 @@
+        qcKey = vName, iconv.OqcName()
+
+        for f in self.filenames:
+            print(" Reading file: ", f)
+            ncd = nc.Dataset(f, 'r')
+
+            source_var_name = {
+                'lat': 'Latitude',
+                'lon': 'Longitude',
+                'icec': 'NASA_Team_2_Ice_Concentration',
+                'icec_qc': 'Flags'
+            }
+            lon = ncd.variables['Longitude'][:]
+            lat = ncd.variables['Latitude'][:]
+            icec = ncd.variables['NASA_Team_2_Ice_Concentration'][:]
+            icec_qc = ncd.variables['Flags'][:]
+            icec_qc = icec_qc.astype(int)
+
+            mask = np.logical_not(icec.mask)
+            lon = lon[mask]
+            lat = lat[mask]
+            icec = icec[mask]
+            icec_qc = icec_qc[mask]
+
+            for i in range(len(lon)):
+                # get date from filename
+                datestart=ncd.getncattr('time_coverage_start')
+                dateend=ncd.getncattr('time_coverage_end')
+                date1=datetime.strptime(datestart,"%Y-%m-%dT%H:%M:%S.%fZ")
+                date2=datetime.strptime(dateend,"%Y-%m-%dT%H:%M:%S.%fZ")
+                avg=date1+(date2-date1)/2
+                
+                locKey = lat[i], lon[i], avg.strftime("%Y-%m-%dT%H:%M:%SZ")
+                self.data[locKey][valKey] = icec[i]*1e-02
+                self.data[locKey][errKey] = 0.1 
+                self.data[locKey][qcKey] = icec_qc[i]
+            ncd.close()
+
+
+def main():
+
+    parser = argparse.ArgumentParser(
+        description=(
+            'Read AMSR-2 sea ice concentration file(s) and convert'
+            ' to a concatenated IODA formatted output file.')
+    )
+    required = parser.add_argument_group(title='required arguments')
+    required.add_argument(
+        '-i', '--input',
+        help="name of icec input file(s)",
+        type=str, nargs='+', required=True)
+    required.add_argument(
+        '-o', '--output',
+        help="name of ioda output file",
+        type=str, required=True)
+    required.add_argument(
+        '-d', '--date',
+        help="base date for the center of the window",
+        metavar="YYYYMMDDHH", type=str, required=True)
+    args = parser.parse_args()
+    fdate = datetime.strptime(args.date, '%Y%m%d%H')
+#
+    VarDims = {
+        'sea_ice_area_fraction': ['nlocs'],
+    }
+
+    # Read in the Ice concentration 
+    icec = iceconc(args.input, fdate)
+
+    # write them out
+    ObsVars, nlocs = iconv.ExtractObsData(icec.data, locationKeyList)
+
+    DimDict = {'nlocs': nlocs}
+    writer = iconv.IodaWriter(args.output, locationKeyList, DimDict)
+
+    VarAttrs = DefaultOrderedDict(lambda: DefaultOrderedDict(dict))
+    VarAttrs[('sea_area_fraction', 'ObsValue')]['units'] = '1'
+    VarAttrs[('sea_area_fraction', 'ObsError')]['units'] = '1'
+    VarAttrs[('sea_area_fraction', 'PreQC')]['units'] = 'unitless'
+    VarAttrs[('sea_area_fraction', 'ObsValue')]['_FillValue'] = 999
```

Comment:
It probably doesn't matter here as there aren't any missing values. But it would be better to read the FillValue from the input file and using that here.

Reply:
Read/add FillValue and units

Reply:
Line 126 should be deleted.  There is no such thing as ```unitless``` in our JEDI Data Convention - so it should simply be skipped.

Reply:
@gthompsnJCSDA line 126 with unitless has been omitted. The units are being read from "units" attribute in the nc file.

Reply:
OK, but please ensure that in the case for which a PreQC variable may have units assigned as ```unitless``` then the converter still should not use those units and instead skip this attribute for this variable.  And that should be the case for any variable in which the input file may contain unitless.  Thanks!